### ISSUE TITLE
Allow for more complicated dependency graph API than groupId

### ISSUE THREAD

**Issue by mikesol** [yigit/android-priority-jobqueue#321]
One use case I am running into in my app is that I have several jobs that can run in parallel but then, when _all_ of them are done, I need to run one job. android-priority-queue is great for chaining individual jobs via `groupId`, but this more complicated type of dependency wiring à la Gradle is not currently supported.  My question - is this in the road map?  I may time to propose a patch if not - let me know!

**Comment by yigit** [comment]
Yes, I've planned this for a while but I'm very busy w/ next Android release these days, don't have any spare time. If you can work on it, that would be awesome and I would happily review any design and code.
